### PR TITLE
Commit aims to get resolutions in the JSON response for Metrics. CDAP-2158

### PR DESCRIPTION
**Pull request by rohanahata**
Commit aims to get resolutions in the JSON response. Also test case coverage.

### CODE REVIEW

**Comment by chtyim** [comment]
It would be good to prefix the commit message with the JIRA issue. Something like:

```
(CDAP-2158) Return metrics resolutions in the JSON response.
 - Some more detail here about the changes.
```

File: cdap-proto/src/main/java/co/cask/cdap/proto/MetricQueryResult.java [hunk 1]
```diff
@@ -44,6 +46,8 @@ public long getEndTime() {
     return series;
   }
 
+  public int getResolution() { return resolution; }
```
Comment (bdmogal) [inline cdap-proto/src/main/java/co/cask/cdap/proto/MetricQueryResult.java]: this should not pass checkstyle. there should be a newline before and after the ``return`` statement.

Btw, you may want to check if your IDE settings are up-to-date. You should get a checkstyle error in IDEA.

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java [hunk 2]
```diff
@@ -471,7 +471,8 @@ private String humanToTagName(String humanTagName) {
     return Lists.newArrayList(Iterables.filter(metricNames, Predicates.notNull()));
   }
 
-  private MetricQueryResult decorate(Collection<MetricTimeSeries> series, long startTs, long endTs) {
+  private MetricQueryResult decorate(Collection<MetricTimeSeries> series, long startTs, long endTs,
+                                     Integer resolution) {
```
Comment (chtyim) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java]: Why takes an `Integer` not `int`?

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java [hunk 3]
```diff
@@ -541,6 +541,24 @@ public void testInterpolate() throws Exception {
...
 
+  @Test
+  public void testResolutionInResponse() throws Exception {
+    long start = 1;
+
+    String url = "/v3/metrics/query?" + getTags("resolutions", "WordCount1", "WordCounter", "splitter") +
+      "&metric=system.reads&resolution=auto&start=" + (start - 1) + "&end="
+      + (start + 36000);
+    MetricQueryResult queryResult = post(url, MetricQueryResult.class);
+    Assert.assertEquals(3600, queryResult.getResolution());
```
Comment (gsps1) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java]: should return 3600s

Comment (gsps1) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java]: also add a test case, where you skip resolution. then check for resolution in batched request as well. you can edit the existing test case if you may for that.

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java [hunk 4]
```diff
@@ -352,7 +352,7 @@ private MetricQueryResult executeQuery(MetricQueryRequest queryRequest) throws E
       endTime = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
     }
 
-    return decorate(queryResult, timeRange.getStart(), endTime);
+    return decorate(queryResult, timeRange.getStart(), endTime, timeRange.getResolutionInSeconds().intValue());
```
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java]: just do, timeRange.gerResolutionInSeconds(), you need to add  the unit - "s" as we talked before.

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java (hunk 3 above)
Comment (rohanahata) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java]: Needed a little clarity on this

**Comment by rohanahata** [comment]
https://builds.cask.co/browse/CDAP-DUT2005/

File: cdap-proto/src/main/java/co/cask/cdap/proto/MetricQueryResult.java [hunk 5]
```diff
@@ -25,11 +25,13 @@
...
   private final long endTime;
   private final TimeSeries[] series;
+  private final String resolution;
 
-  public MetricQueryResult(long startTime, long endTime, TimeSeries[] series) {
+  public MetricQueryResult(long startTime, long endTime, TimeSeries[] series, int resolution) {
     this.startTime = startTime;
     this.endTime = endTime;
     this.series = series;
+    this.resolution = resolution + "s";
```
Comment (gsps1) [inline cdap-proto/src/main/java/co/cask/cdap/proto/MetricQueryResult.java]: String.valueOf(resolution) + "s"

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java (hunk 4 above)
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java]: can remove intValue()

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java [hunk 6]
```diff
@@ -329,7 +335,27 @@ public void testInvalidRequest() throws Exception {
       this.metrics = metrics;
       this.groupBy = groupBy;
       this.timeRange = timeRange;
+      this.resolution = resolution;
```
Comment (gsps1) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java]: do you need this ? i don't see resolution in the constructor params ?

**Comment by gsps1** [comment]
LGTM

**Pull request merged**